Sources/Disposable.swift

```diff
@@ -38,6 +38,13 @@ extension UnsafeAtomicState where State == DisposableState {
 	}
 }
 
+internal final class NopDisposable: Disposable {
+	static let shared = NopDisposable()
+	var isDisposed = false
+	func dispose() {}
+	private init() {}
+}
```

Comment:
This breaks the contract of `Disposable` by not changing `isDisposed` when you call `dispose`. It may be fine in practice, but this makes me very wary.

I think we'd be better off bringing back the implementation of `SimpleDisposable` and using that instead.